tests/test_runner/test_discover_runner.py

```diff
@@ -71,6 +79,65 @@ def test_pattern(self):
 
         self.assertEqual(count, 1)
 
+    @skipUnless(PY37, "unittest -k option requires Python 3.7 and later")
+    def test_name_patterns(self):
```

Comment:
Consider using a single test and `subTest()` to avoid the same boilerplate in each test.